internal/resolution/variablesources/installed_package.go

```diff
@@ -83,3 +90,28 @@ func legacySemanticsSuccessors(allBundles []*catalogmetadata.Bundle, installedBu
 
 	return upgradeEdges, nil
 }
+
+// semverSuccessors returns successors based on Semver.
+// Successors will not include versions outside the major version of the
+// installed bundle as major version is intended to indicate breaking changes.
+func semverSuccessors(allBundles []*catalogmetadata.Bundle, installedBundle *catalogmetadata.Bundle) ([]*catalogmetadata.Bundle, error) {
+	currentVersion, err := installedBundle.Version()
+	if err != nil {
+		return nil, err
+	}
+
+	// Based on current version create a caret range comparison constraint
+	// to allow only minor and patch version as successors and exclude current version.
+	constraintStr := fmt.Sprintf("^%s, != %s", currentVersion.String(), currentVersion.String())
```

Comment:
Based on [this doc](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004), a caret range will prevent the left-most non-zero digit in the `[major, minor, patch]` tuple.
In other words, this allows:
>1. patch and minor updates for versions 1.0.0 and above
>2. patch updates for versions 0.X >=0.1.0
>3. no updates for versions 0.0.X.

What happens in cases 2 and 3? I think we'll need good documentation to address these cases.

Case 2 doesn't seem too concerning given this comment from the doc linked earlier:
>Many authors treat a 0.x version as if the x were the major "breaking-change" indicator.

Case 3 seems more concerning, I suspect that many people packaging an operator for the first time will use version `0.0.1` and fail to upgrade it.

Reply:
Beauty!